### DISCUSSION TITLE
Tracker Config for Fast Moving Objects

### DISCUSSION THREAD

**Discussion by SaumyaRaj188** [ultralytics/ultralytics#13404] [Q&A]
I have a project to track cars across 5 cameras, My GTX 1650 takes about 280ms to process all 5 frames. So when running on live feed no ID is assigned to the cars sometimes( especially fast moving cars) as it skips through many frames.
I am trying to pass a custom_tracker.yaml file but I am not sure what each parameter does. 

And if there is some other better method to do this task please specify.

**Comment by glenn-jocher** [comment]
Hello,

For tracking fast-moving objects like cars across multiple cameras, adjusting your tracker configuration can indeed help improve performance and accuracy. Here are a few suggestions:

1. **Increase the Frame Rate**: If possible, increase the processing speed to handle more frames per second. This might involve optimizing your model or hardware upgrades if feasible.

2. **Adjust Tracker Configurations**: In your `custom_tracker.yaml`, you can adjust parameters such as `track_high_thresh` to manage the confidence threshold for detections considered for tracking. Lowering this might help in retaining IDs for fast-moving cars.

3. **Use a Robust Tracker**: Consider using the ByteTrack tracker (`bytetrack.yaml`), which is known for its effectiveness in handling occlusions and fast-moving objects.

4. **Batch Processing**: If latency is less of a concern, processing the frames in batches rather than frame-by-frame might give the tracker more context for better ID persistence.

For detailed explanations of each parameter in the tracker configuration file, you can refer to the comments within the YAML file itself, which typically describe their roles and impacts.

Implementing these adjustments should help in managing the tracking of fast-moving objects more effectively.

**Comment by SaumyaRaj188** [comment]
I have found a solution, custom training helped a lot in increasing the confidence of detection, plus increasing the match_threshold to a high value (10 in my case) helped with fast motion it keeps the track going. Also most important thing was implementing a masking over the frame, i blacken out the surrounding pixels where it was making wrong prediction and only kept the area where the car i want to track will pass through, also added some masked images in the dataset while training, it works well now.

**Reply by pderrenger** [reply]
@SaumyaRaj188 i'm glad to hear that custom training and adjusting the match_threshold have improved your tracking performance. Implementing masking to focus on relevant areas is a smart approach, especially for fast-moving objects. If you encounter any further issues or need additional assistance, feel free to reach out. Keep up the great work!
